### DISCUSSION TITLE
Show the diff for "The latest configuration has not been applied"

### DISCUSSION THREAD

**Discussion by Geczy** [coollabsio/coolify#2405] [Feature Requests]
to reproduce, visit a project that's "outdated" ? and see this message, but no clue as to what exactly has changed, and why we need to redeploy

```md
The latest configuration has not been applied
Please restart (or redeploy) to apply the new configuration.
```

**Comment by nev1d** [comment]
+1. I have exactly the same problem and I have no idea where these changes happened

**Comment by Jihad** [comment]
+1. I believe i have a service stuck on this. Restarted multiple times with pulling images again. Same.

**Comment by steff-mat** [comment]
Please mention the Coolify version, the service(s) which are affected and if it was deployed via Applications (Git/Docker Based) or from the Databases/Services list.

**Reply by Jihad** [reply]
@steff-mat Thanks! I might not be original poster, but with me, i'm on: 
- Coolify [v4.0.0-beta.367](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.367)
- Happens on Services list: Grafana, Metabase, n8n-with-postgresql
- Doesn't happen on: cloudbeaver (from service list)

restarted them couple times, still getting it. Even with pull latest & restart. Not sure if a can get logs or do manual check of configurations vs. current to know what's going on.

**Comment by jhgeluk** [comment]
I started to experience the same on:
[v4.0.0-beta.370](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.370)

**Comment by sergiotapia** [comment]
Same on [v4.0.0-beta.380](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.380) seems stuck there?

**Comment by Gog0** [comment]
Same for me on [v4.0.0-beta.394](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.394) with a newly installed firefly app. It started after I made database public I believe.

**Reply by Jihad** [reply]
Same! On [v4.0.0-beta.420.6](https://github.com/coollabsio/coolify/releases/tag/v4.0.0-beta.420.6)

Firefly as well.

**Comment by djsisson** [comment]
Just press save and then restart

It computes a hash of all the various settings and fields for a project and compares that with a saved value

So when upgrading versions if a new db column is added or a default setting is changed, the hash now won't match, pressing save will fix the warning.
